enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/protocol/Protocol.java

```diff
@@ -53,13 +63,68 @@ public String canonicalName()
         @Override
         public String identifier()
         {
-            return this.identifier.canonicalName();
+            return identifier.canonicalName();
+        }
+
+        @Override
+        public int version()
+        {
+            return version;
+        }
+    }
+
+    interface ModifierProtocol extends Protocol
+    {
+        /**
+         * Should be human readable in a comma separated list
+         */
+        String friendlyName();
+    }
+
+    enum ModifierProtocolIdentifier implements Identifier<ModifierProtocol>
+    {
+        COMPRESSION,
+        GRATUITOUS_OBFUSCATION;
+
+        @Override
+        public String canonicalName()
+        {
+            return name().toLowerCase();
+        }
+    }
+
+    enum ModifierProtocols implements ModifierProtocol
+    {
+        // TODO Snappy on Raft breaks CoreToCoreCopySnapshotIT
+        COMPRESSION_NONE( ModifierProtocolIdentifier.COMPRESSION, 1, "Uncompressed" );
+
+        private final int version;
+        private final ModifierProtocolIdentifier identifier;
+        private final String friendlyName;
+
+        ModifierProtocols( ModifierProtocolIdentifier identifier, int version, String friendlyName )
```

Comment:
This is very much different than what was done for application protocols. Not sure version makes sense anymore as a name for this, or even identifier. Identifier seems to be a class of modifiers while the version decided the particular implementation. Will friendly name be sent over the wire? I think that is important for the application protocols and makes it possible to still print and complain on something reasonable on an instance which lacks support, instead of just "not supporting compression variant 3".

Reply:
The model is a little odd. I think it makes sense to revisit when implementing configuration - these protocols need to be identified somehow.